app/models/invoice/data_presenter.rb

```diff
@@ -72,7 +72,15 @@ def shipment_adjustment
 
     # contains limited information about the shipment
     def shipment
-      shipment_adjustment.adjustable
+      shipment_adjustment&.adjustable || null_shipment
+    end
+
+    def null_shipment
+      Struct.new(
+        :amount,
+        :included_tax_total,
+        :additional_tax_total,
+      ).new(0, 0, 0)
     end
```

Comment:
It looks like this is used to avoid errors in the below two methods, which would return "$0". Is it necessary to display "$0"  on the invoice here? 

I wonder if it would be better to simply display nothing. Eg this would return `nil`:

```
    def display_shipment_amount_without_taxes
      Spree::Money.new(shipment.amount - shipment.included_tax_total, currency:) if shipment.present?
    end
```